daemon/algod/api/server/v2/handlers.go

```diff
@@ -585,35 +590,57 @@ func (v2 *Handlers) GetProof(ctx echo.Context, round uint64, txid string, params
 		return notFound(ctx, err, "protocol does not support Merkle proofs", v2.Log)
 	}
 
+	hashtype := "sha512_256" // default hash type for proof
+	if params.Hashtype != nil {
+		hashtype = *params.Hashtype
+	}
+	if hashtype == "sha256" && !proto.EnableSHA256TxnCommitmentHeader {
+		return badRequest(ctx, err, "protocol does not support sha256 vector commitment proofs", v2.Log)
+	}
+
 	txns, err := block.DecodePaysetFlat()
 	if err != nil {
 		return internalError(ctx, err, "decoding transactions", v2.Log)
 	}
 
 	for idx := range txns {
-		if txns[idx].Txn.ID() == txID {
-			tree, err := block.TxnMerkleTree()
+		if txns[idx].ID() != txID {
+			continue // skip
+		}
+
+		var tree *merklearray.Tree
+		var stibhash crypto.Digest
+
+		switch hashtype {
+		case "sha256":
+			tree, err = block.TxnMerkleTreeSHA256()
 			if err != nil {
-				return internalError(ctx, err, "building Merkle tree", v2.Log)
+				return internalError(ctx, err, "building Vector Commitment (SHA256)", v2.Log)
 			}
-
-			proof, err := tree.ProveSingleLeaf(uint64(idx))
+			stibhash = block.Payset[idx].HashSHA256()
+		case "sha512_256":
+			tree, err = block.TxnMerkleTree()
 			if err != nil {
-				return internalError(ctx, err, "generating proof", v2.Log)
+				return internalError(ctx, err, "building Merkle tree", v2.Log)
 			}
+			stibhash = block.Payset[idx].Hash()
+		default:
+			return badRequest(ctx, err, "unsupported hash type", v2.Log)
+		}
 
-			stibhash := block.Payset[idx].Hash()
-
-			response := generated.ProofResponse{
-				Proof:     proof.GetConcatenatedProof(),
-				Stibhash:  stibhash[:],
-				Idx:       uint64(idx),
-				Treedepth: uint64(proof.TreeDepth),
-				Hashtype:  proof.HashFactory.HashType.String(),
```

Comment:
So you removed hashtype from the response, but it is still listed as required in the ProofResponse YAML spec...

Reply:
[fixed here](https://github.com/algorand/go-algorand/pull/3985)